### PR TITLE
[option] Use ml_option_get to get value from ml_option_h

### PR DESCRIPTION
**Pull request by anyj0527**
- Use ml_option API rather than accessing internal structure.
- Declare ml_info and related enum and structures. And add internal functions for create, destroy, set, get.
- Replace ml_option_s with the new internal structure in ml_option API.

### CODE REVIEW

File: c/src/ml-api-service-query-client.c [hunk 1]
```diff
@@ -98,45 +94,42 @@ ml_service_query_create (ml_option_h option, ml_service_h * h)
...
+  if (ML_ERROR_NONE == ml_option_get (option, "timeout", &value))
+    g_string_append_printf (tensor_query_client_prop, " timeout=%u ",
+        *((guint *) value));
+
+  if (ML_ERROR_NONE != ml_option_get (option, "caps", &value)) {
     g_string_free (tensor_query_client_prop, TRUE);
     _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
         "The option 'caps' must be set before call ml_service_query_create.");
   }
+  caps = g_strdup ((gchar *) value);
```
Comment (jaeyun-jung) [inline c/src/ml-api-service-query-client.c:132]: inside of the block? line 127

Comment (anyj0527) [reply c/src/ml-api-service-query-client.c:132]: L127 has different expression with other if lines, cause `caps` is necessary for this API.

File: c/src/ml-api-common.c [hunk 2]
```diff
@@ -1287,25 +1287,117 @@ ml_strerror (int errnum)
...
+{
+  ml_info_value_s *info_value;
+
+  if (!info || !key || !value)
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'info', 'key' or 'value' is NULL. It should be a valid ml_info, key and value.");
+
+  info_value = g_hash_table_lookup (info->table, key);
+  if (!info_value)
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
```
Comment (jaeyun-jung) [inline c/src/ml-api-common.c:1386]: Let's reduce the error message, sometimes business division complains unnecessary error logs.
```if (!info_value) return ML_ERROR_INVALID_PARAMETER;```

File: c/src/ml-api-service-query-client.c (hunk 1 above)
Comment (jaeyun-jung) [reply c/src/ml-api-service-query-client.c:132]: understood

**Review by songgot — APPROVED** [review]
LGTM :+1:

File: c/src/ml-api-common.c (hunk 2 above)
Comment (anyj0527) [reply c/src/ml-api-common.c]: Thanks, I updated the code PTAL.

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**